File: codeforces/41.hit_the_lottery/hit_the_lottery.py
# Author: José Rodolfo (jric2002)
import math
n = int(input())
min_num_bills = 0
dollars = [1, 5, 10, 20, 100]
num_bills = len(dollars)
while (n != 0):
  i = num_bills - 1
  while (i >= 0):
    if (n >= dollars[i]):
      min_num_bills += math.floor(n / dollars[i])
      n = (n % dollars[i])
    i -= 1
print(min_num_bills)

File: codeforces/41.hit_the_lottery/hit_the_lottery.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned long int n, min_num_bills = 0;
  unsigned short int dollars[5] = {1, 5, 10, 20, 100};
  cin >> n;
  while (n != 0) {
    for (short int i = (sizeof(dollars)/sizeof(unsigned short int) - 1); i >= 0; i--) {
      if (n >= dollars[i]) {
        min_num_bills += (n / dollars[i]);
        n = (n % dollars[i]);
      }
    }
  }
  cout << min_num_bills << endl;
  return 0;
}